commands/clients/create.js

```diff
@@ -15,9 +15,15 @@ function * run (context, heroku) {
       redirect_uri: url
     }
   })
-  let client = yield cli.action(`Creating ${context.args.name}`, request)
-  cli.log(`HEROKU_OAUTH_ID=${client.id}`)
-  cli.log(`HEROKU_OAUTH_SECRET=${client.secret}`)
+  if (context.flags.shell) {
+    let client = yield request
+    cli.log(`HEROKU_OAUTH_ID=${client.id}`)
+    cli.log(`HEROKU_OAUTH_SECRET=${client.secret}`)
+  } else {
+    let client = yield cli.action(`Creating ${context.args.name}`, request)
```

Comment:
you could show this action message even when using `--shell` since it prints to `stderr`. I could go either way.